File: ApacheStormPrimeJava/src/PrimeTopology.java
import org.apache.storm.Config;
import org.apache.storm.LocalCluster;
import org.apache.storm.topology.TopologyBuilder;

public class PrimeTopology {

    public static void main(String[] args) throws Exception {
        TopologyBuilder builder = new TopologyBuilder();
        builder.setSpout("spout", new PrimeSpout());
        builder.setBolt("bolt", new PrimeBolt()).shuffleGrouping("spout");

        Config config = new Config();
        LocalCluster cluster = new LocalCluster();
        cluster.submitTopology("topology", config, builder.createTopology());

        Thread.sleep(5000);

        cluster.killTopology("topology");
        cluster.shutdown();
    }
}

File: ApacheStormPrimeScala/src/main/scala/com/prime/PrimeTopology.scala
package com.prime

import org.apache.storm.{Config, LocalCluster, StormSubmitter}
import org.apache.storm.topology.TopologyBuilder
import org.apache.storm.utils.Utils

object PrimeTopology extends App {
  val builder: TopologyBuilder = new TopologyBuilder
  builder.setSpout("spout", new PrimeSpout)
  builder.setBolt("bolt", new PrimeBolt).shuffleGrouping("spout")
  System.setProperty("storm.jar", "/Users/a.azad/IdeaProjects/apache-storm-prime/ApacheStormPrimeScala/target/scala-2.12/primecheck_2.12-1.0.jar")

  val config: Config = new Config
  // val cluster: LocalCluster = new LocalCluster
  StormSubmitter.submitTopology("topology", config, builder.createTopology())

  Runtime.getRuntime.addShutdownHook(new Thread() {
    override def run(): Unit = {
      println("Shutting down storm topology of prime numbers ...")
    }
  })

  Utils.sleep(30000)
  System.exit(0)
}
